contrib/parseq-retry/src/main/java/com/linkedin/parseq/retry/ResultClassification.java

```diff
@@ -0,0 +1,35 @@
+package com.linkedin.parseq.retry;
+
+
+/**
+ * Definitions of the supported result classifications and the default classification function.
+ *
+ * @author Oleg Anashkin (oleg.anashkin@gmail.com)
+ */
+public enum ResultClassification {
+  /** The classification of results that will not interrupt the retry operation. */
+  ACCEPTABLE,
+
+  /** The classification of results that will interrupt the retry operation. */
+  UNACCEPTABLE;
+
+  /** Error classification associated with this result. */
+  protected ErrorClassification _status = ErrorClassification.RECOVERABLE;
+
+  /**
+   * @return Error classification associated with this result.
+   */
+  public ErrorClassification getStatus() {
+    return _status;
+  }
+
+  /**
+   * Associate the error classification with this result. It controls the behavior of retry operation.
+   *
+   * @param status Error classification.
+   */
+  public ResultClassification setStatus(ErrorClassification status) {
```

Comment:
Because that would need three unacceptable values: UNACCEPTABLE_FATAL, UNACCEPTABLE_RECOVERABLE and UNACCEPTABLE_SILENTLY_RECOVERABLE, essentially tightly coupling it with ErrorClassification enums. I didn't want to do that. I'll see what I can do about avoiding global effect.